### PR TITLE
Add in.bharatkalluri.splash

### PR DESCRIPTION
**Pull request by BharatKalluri**
Source Code: https://gitlab.gnome.org/Bharatkalluri/splash

### CODE REVIEW

File: in.bharatkalluri.splash.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+    "app-id" : "in.bharatkalluri.splash",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.34",
```
Comment (LeandroStanger) [inline in.bharatkalluri.splash.json:4]: Update runtime to 3.36
```suggestion
    "runtime-version" : "3.36",
```

File: in.bharatkalluri.splash.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        "--socket=fallback-x11",
+        "--socket=wayland",
+
+        "--filesystem=~/.cache",
+        "--filesystem=~/Pictures",
+
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
```
Comment (LeandroStanger) [inline in.bharatkalluri.splash.json:19]:
```suggestion
```

Comment (BharatKalluri) [reply in.bharatkalluri.splash.json:19]: Splash needs to talk to dconf (`org.gnome.desktop.background` gsetting) to change the wallpaper. If I drop these lines, I think I won't be able to change the wallpaper.

Comment (bilelmoussaoui) [reply in.bharatkalluri.splash.json:19]: You can use the wallpaper portal for that

File: in.bharatkalluri.splash.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    "runtime-version" : "3.36",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "splash",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+
+        "--filesystem=~/.cache",
```
Comment (bilelmoussaoui) [inline in.bharatkalluri.splash.json:13]:
```suggestion
```
You don't need access to this. You already have access to it.

File: in.bharatkalluri.splash.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        "--filesystem=~/Pictures",
```
Comment (bilelmoussaoui) [inline in.bharatkalluri.splash.json:14]:
```suggestion
        "--filesystem=xdg-pictures",
```

File: in.bharatkalluri.splash.json (hunk 2 above)
Comment (bilelmoussaoui) [reply in.bharatkalluri.splash.json:18]: Not a blocker though, as there's no easy and nice way to do that without using d-bus calls yourself.

**Comment by bilelmoussaoui** [comment]
bot, build in.bharatkalluri.Splash

**Comment by bilelmoussaoui** [comment]
bot, build in.bharatkalluri.splash

File: in.bharatkalluri.splash.json (hunk 2 above)
Comment (BharatKalluri) [reply in.bharatkalluri.splash.json:18]: I see, in that case let us merge this if everything else looks okay to you. I will create an issue for this in my repo.

**Comment by BharatKalluri** [comment]
Looks like the wallpaper is not being set. I am trying to figure out what went wrong..

**Comment by bilelmoussaoui** [comment]
that's why it doesn't work https://gitlab.gnome.org/Bharatkalluri/splash/-/blob/master/src/services/wallpaper_service.py#L12-16 you should use glib's api to get the proper cache dir instead of hardcoding it.

**Comment by BharatKalluri** [comment]
bot, build in.bharatkalluri.splash

**Comment by BharatKalluri** [comment]
Thanks for the tip @bilelmoussaoui , it works as expected!

**Pull request closed without merging**